Userland/Libraries/LibFileSystem/FileSystem.cpp

```diff
@@ -47,12 +47,15 @@ ErrorOr<String> real_path(StringView path)
         return Error::from_errno(ENOENT);
 
     DeprecatedString dep_path = path;
-    auto* real_path = realpath(dep_path.characters(), nullptr);
+    auto* real_path_ptr = realpath(dep_path.characters(), nullptr);
 
-    if (!real_path)
+    if (!real_path_ptr)
         return Error::from_syscall("realpath"sv, -errno);
 
-    return TRY(String::from_deprecated_string({ real_path }));
+    auto real_path_string = TRY(String::from_deprecated_string({ real_path_ptr }));
```

Comment:
`String::from_utf8({ real_path_ptr, strlen(real_path_ptr) })` here, let's not have a third allocation :)